toolchain/ltrace.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+backend = SoftwareManager()
+dist = distro.detect()
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop
+        a library function check_repo() should be implemented.
+        """
+
+        os_deps = ['git', 'wget', 'rpm', 'autoconf',
+                   'automake', 'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs = os_deps + sles_deps
+
+        elif dist.name == 'redhat':
+            rhel_deps = ['elfutils-devel', 'elfutils-libelf-devel',
+                         'elfutils-libelf', 'elfutils-libs', 'libtool-ltdl']
+            pkgs = os_deps + rhel_deps
+
+        elif dist.name == 'ubuntu':
+            ubuntu_deps = ['elfutils', 'libelf-dev', 'libtool',
+                           'libelf1', 'librpmbuild3', 'binutils-dev']
+            pkgs = os_deps + ubuntu_deps
+        else:
+            print "Unsupported OS!"
+
+        i = 0
+        while i < len(pkgs):
+            if backend.install(pkgs[i]):
+                print "%s installed successfull" % pkgs[i]
+            else:
+                print "Error: Check repository !"
+                exit(1)
+            i = i + 1
+
+        """
+        clone the  ltrace tests from the GIT repo
+        Source: git clone git://git.debian.org/git/collab-maint/ltrace.git
+        """
+
+        git.get_repo('git://git.debian.org/git/collab-maint/ltrace.git',
+                     destination_dir=self.srcdir + '/ltrace')
+        ltrace_version = os.path.basename("ltrace")
+        os.chdir(self.srcdir)
+
+        self.srcdir = os.path.join(self.srcdir, ltrace_version)
+        os.chdir(self.srcdir)
+        process.run('./autogen.sh')
+        process.run('./configure')
+        build.make(self.srcdir)
+        try:
+            process.run('which gcc')
+        except error.CmdError:
+            raise NotAvailableError('gcc is required by this job and is '
+                                    'not available on the system')
+
+    def test(self):
+        os.chdir(self.srcdir)
+        build.make(self.srcdir, extra_args='check')
```

Comment:
Hi ldoktor

I have incorporated all the review comments which are  mentioned.
Ran the ltrace.py through avocado.
Observations.
While  performing  the "make check" - it stays there in make check for longer duration - It is not coming out from this loop.
After that added the code to check the *.log files.
It is not entering the next lines of code to check .
How to go about this statements to execute and make the test is pass/fail . Please suggest.